shared/src/main/java/uk/gov/di/authentication/shared/services/DynamoAuthenticationAttemptsService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.di.authentication.shared.services;
+
+import uk.gov.di.authentication.shared.entity.AuthenticationAttempts;
+import uk.gov.di.authentication.shared.helpers.NowHelper;
+
+import java.time.Clock;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+public class DynamoAuthenticationAttemptsService extends BaseDynamoService<AuthenticationAttempts> {
+
+    public DynamoAuthenticationAttemptsService(ConfigurationService configurationService) {
+        super(AuthenticationAttempts.class, "authentication-attempts", configurationService);
+        new NowHelper.NowClock(Clock.systemUTC());
+    }
+
+    public void addCode(String attemptIdentifier, long ttlSeconds, String code) {
+        long ttlEpochSeconds =
+                NowHelper.nowPlus(ttlSeconds, ChronoUnit.SECONDS).toInstant().getEpochSecond();
+
+        var authenticationAttempt =
+                get(attemptIdentifier)
+                        .orElse(new AuthenticationAttempts())
+                        .withAttemptIdentifier(attemptIdentifier)
+                        .withCode(code)
+                        .withTimeToLive(ttlEpochSeconds);
+
+        update(authenticationAttempt);
+    }
+
+    public void createOrIncrementCount(String attemptIdentifier, long ttl) {
```

Comment:
Does `createOrIncrementCount ` also need to accept and store the `authenticationType/Method`?